File: vueprogect/src/api/gridCode.ts
import request from '@/utils/request'

// 获取网格码列表
export function getGridCodeList(params: {
  gridCodeValue?: string
  status?: string
  currentPage: number
  pageSize: number
}) {
  return request({
    url: '/api/gridCode/list',
    method: 'get',
    params
  })
}

// 获取网格码详情
export function getGridCodeDetail(id: number) {
  return request({
    url: `/api/gridCode/${id}`,
    method: 'get'
  })
}

// 创建网格码
export function createGridCode(data: {
  gridCodeValue: string
  status?: string
  hashValue: string
  depositRecordId: string
}) {
  return request({
    url: '/api/gridCode',
    method: 'post',
    data
  })
}

// 更新网格码
export function updateGridCode(id: number, data: {
  status?: string
  hashValue?: string
  depositRecordId?: string
}) {
  return request({
    url: `/api/gridCode/${id}`,
    method: 'put',
    data
  })
}

// 删除网格码
export function deleteGridCode(id: number) {
  return request({
    url: `/api/gridCode/${id}`,
    method: 'delete'
  })
}

// 验真网格码
export function verifyGridCode(id: number) {
  return request({
    url: `/api/gridCode/verify/${id}`,
    method: 'post'
  })
}

File: syz-user/src/main/java/com/cuco/syzuser/entity/GridCode.java
package com.cuco.syzuser.entity;

import lombok.Data;

@Data
public class GridCode {
    private Long gridCodeId;        // 表内唯一标识
    private String gridCodeValue;   // 网格码唯一值
    private Long userId;            // 关联用户ID（逻辑外键）
    private String status;          // 状态：VALID-有效、INVALID-失效
    private String hashValue;       // 区块链存证哈希值
    private String depositRecordId; // 区块链存证记录ID
}
